### ISSUE TITLE
Error executing query 3 level deep where variable in @neo4j/graphql@3.16.0

### ISSUE THREAD

**Issue by alvaro-lara-lll** [neo4j/graphql#2871]
**Describe the bug**
We have a query where the where variable is implementing a one-to-one relationship to a second level type and then to a one-to-many relationship to a third level type: 
```
 "where": {
    "secondLevel": {
      "thirdLevel": {
        "id": "test"
      }
    }
  }
```
This query was working fine in @neo4j/graphql@3.14.1.
We first experienced the error with @neo4j/graphql@3.15.1, but it also occurring in @neo4j/graphql@3.16.0.

**Type definitions**
```
  type FirstLevel {
    id: ID! @id(unique: true)
    secondLevel: SecondLevel! @relationship(type: "HAS_SECOND_LEVEL", direction: OUT)
    createdAt: DateTime! @timestamp(operations: [CREATE])
  }

  type SecondLevel {
    id: ID! @id(unique: true)
    thirdLevel: [ThirdLevel!]! @relationship(type: "HAS_THIRD_LEVEL", direction: OUT)
    createdAt: DateTime! @timestamp(operations: [CREATE])
  }

  type ThirdLevel {
    id: ID! @id(unique: true)
    createdAt: DateTime! @timestamp(operations: [CREATE])
  }
```

**To Reproduce**

1. Run the following Query:
Query:
```
query FirstLevels($where: FirstLevelWhere) {
  firstLevels(where: $where) {
    id
  }
}

Variables:
{
  "where": {
    "secondLevel": {
      "thirdLevel": {
        "id": "test"
      }
    }
  }
}
```
2. See error:
```
"EXISTS is only valid in a WHERE clause as a standalone predicate or as part of a boolean expression (AND / OR / NOT) (line 2, column 89 (offset: 114))\n\"WHERE single(this0 IN [(this)-[:HAS_SECOND_LEVEL]->(this0:`SecondLevel`) | this0] WHERE EXISTS {\"\n ^"
```

**Expected behavior**
A list of FirstLevel type nodes or empty list (if there are no nodes matching the where criteria).

**System (please complete the following information):**
* Version: [e.g. @neo4j/graphql@3.16.0]

**Additional context**
It seems the error is related to the "single" predicate function. The generated cypher query in 3.14.1 was not using this function.

- This is the generated cypher query with @neo4j/graphql@3.14.1 :
```
MATCH (this:`FirstLevel`)
WHERE EXISTS {
    MATCH (this)-[:HAS_SECOND_LEVEL]->(this0:`SecondLevel`)
    WHERE EXISTS {
        MATCH (this0)-[:HAS_THIRD_LEVEL]->(this1:`ThirdLevel`)
        WHERE this1.id = $param0
    }
}


RETURN this { .id } AS this
```

- This is the generated cypher query with @neo4j/graphql@3.16.0:
```
MATCH (this:`FirstLevel`)
WHERE single(this0 IN [(this)-[:HAS_SECOND_LEVEL]->(this0:`SecondLevel`) | this0] WHERE EXISTS {
    MATCH (this0)-[:HAS_THIRD_LEVEL]->(this1:`ThirdLevel`)
    WHERE this1.id = $param0
})


RETURN this { .id } AS this
 ```

**Comment by neo4j-team-graphql** [comment]
Many thanks for raising this bug report @alvaro-lara-lll. :bug: We will now attempt to reproduce the bug based on the steps you have provided.

Please ensure that you've provided the necessary information for a minimal reproduction, including but not limited to:
* Type definitions
* Resolvers
* Query and/or Mutation (or multiple) needed to reproduce

If you have a support agreement with Neo4j, please link this GitHub issue to a new or existing Zendesk ticket.

Thanks again! :pray:

**Comment by neo4j-team-graphql** [comment]
Many thanks for raising this bug report @alvaro-lara-lll. :bug: We will now attempt to reproduce the bug based on the steps you have provided.

Please ensure that you've provided the necessary information for a minimal reproduction, including but not limited to:
* Type definitions
* Resolvers
* Query and/or Mutation (or multiple) needed to reproduce

If you have a support agreement with Neo4j, please link this GitHub issue to a new or existing Zendesk ticket.

Thanks again! :pray:

**Comment by tbwiss** [comment]
Hi @alvaro-lara-lll!
Thanks for reaching out and reporting this bug!
I was able to reproduce it, yet only with a Neo4j version 4.4 database. No error occurs on a Neo4j 5.X database.
Can you confirm that you encountered this error with a Neo4j 4.X (4.3 or 4.4) version?

**Comment by neo4j-team-graphql** [comment]
We've been able to confirm this bug using the steps to reproduce that you provided - many thanks @alvaro-lara-lll! :pray: We will now prioritise the bug and address it appropriately.

**Comment by neo4j-team-graphql** [comment]
This bug report has been assigned high priority to fix. If you wish to contribute a fix, please branch from `master` and submit your PR with the base set to `master`. Thanks!

**Comment by alvaro-lara-lll** [comment]
Thank you Thomas. I saw that the issue was resolved as part of #2877. Just
to answer your question I did encounter the error with a Neo4j 4.4 database.
